File: TP02/company.js
import Employee from './employee'

class Company {
    constructor(employees, name, cuit){
        this.employees = employees;
        this.name = '',
        this.cuit = '',
        this.liquidation();
    }

    /**GETTERS */
    getEmployees(){
        return this.employees;
    }

    getName(){
        return this.name;
    }

    getCuit(){
        return this.cuit;
    }

    getRetirementContributions(){
        return this.retirementContributions;
    }
    /**SETTERS */
    setEmployees(employees){
        this.employees = employees;
    }

    setName(name){
        this.name = name;
    }

    setCuit(cuit){
        this.cuit = cuit;
    }

    setRetorementContributions(retorementContributions){
        this.retirementContributions = retorementContributions;
    }

    /**FUNCTIONS */
    totalNetSalaries(employees){
        var result = 0;
        for (employee in employees){
            result = result + employee.netSalary();
        }
        return result;
    }

    totalGrossSalaries(employees){
        var result = 0;
        for (employee in employees){
            result = result + employee.grossSalary();
        }
        return result;
    }

    totalWithHoldings(employees){
        var result = 0;
        for (employee in employees){
            result = result + employee.withHoldings();
        }
        return result;
    }

    liquidation(employees){
        var date = new Date().toISOString().substr(0, 10);
        for(employee in employees){
            var concept = new Concept(employee.grossSalary())
            var retorementContributions = new RetorementContributions(employee.getName(), employee.getDirection(), date, employee.grossSalary(),employee.netSalary(), concept);
            this.retirementContributions.push(retorementContributions);
    }
}
}

File: TP02 - vJAVA/src/unq/Company.java
package unq;

import java.util.ArrayList;

public class Company {
	private ArrayList<Employee> employees;
	private String name;
	private Integer cuit;
	private ArrayList<RetirementContributions> retirementContributions;
	
	public Company(String name, Integer cuit) {
		super();
		this.name = name;
		this.cuit = cuit;
	}

	public ArrayList<Employee> getEmployees() {
		return employees;
	}

	public void setEmployees(ArrayList<Employee> employees) {
		this.employees = employees;
	}
	
	public void setEmployee(Employee employee) {
		this.employees.add(employee);
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public Integer getCuit() {
		return cuit;
	}

	public void setCuit(Integer cuit) {
		this.cuit = cuit;
	}

	public ArrayList<RetirementContributions> getRetirementContributions() {
		return retirementContributions;
	}

	public void setRetirementContributions(ArrayList<RetirementContributions> retirementContributions) {
		this.retirementContributions = retirementContributions;
	}
	
	/**METHODS*/
	public Integer totalNetSalaries() {
		Integer res = 0;
		for (Employee employees2 : employees) {
			res = res + employees2.netSalary();
		}
		return res;
	}
	
	public Integer totalGrossSalaries() {
		Integer res = 0; 
		for (Employee employees2 : employees) {
			res = res + employees2.grossSalary();
		}
		return res;
	}
	
	public Integer totalWithholdings() {
		Integer res = 0;
		for (Employee employees2 : employees) {
			res = res + employees2.withHoldings();
			}
		return res;
	}
	
	public void liquidation(ArrayList<Employee> employees) {
		
	}
}
